rust/lance/src/dataset.rs

```diff
@@ -1485,6 +1487,28 @@ impl Dataset {
         *self = self.checkout_version(latest_version).await?;
         Ok(())
     }
+
+    /// Run a SQL query against the dataset.
+    pub async fn sql(
```

Comment:
There are various config options when running an SQL query (e.g. how much RAM to use before spilling?  Is spilling even enabled?  How many threads should be used for compute operations?

Do you think this method will grow to address these options?  If so we might want to add a basic `SqlOptions` struct which today can just be `with_row_id` and `with_row_addr`